src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -149,9 +168,146 @@ public static Colony createColony(@NotNull final World w, final BlockPos pos, @N
+
+        final int maxRange = range * 2 + buffer;
+
+        for(int i = chunkX - maxRange; i <= chunkX + maxRange; i++)
+        {
+            for(int j = chunkZ - maxRange; j <= chunkZ + maxRange; j++)
+            {
+                final BlockPos pos = new BlockPos(i, 0, j);
+                if(i >= chunkX - range && j >= chunkZ - range && i <= chunkX + range && j <= chunkZ + range)
+                {
+                    ownedChunks.put(new ChunkPos(i,j,dimension), new ChunkLoadStorage(id, pos, add, dimension));
+                }
+                else
+                {
+                    closeChunks.add(new ChunkLoadStorage(id, pos, add, dimension));
+                }
+            }
+        }
+        removedElements = RELOAD_AFTER_X_ELEMENTS;
+    }
+
+    /**
+     * Load each chunk on world tick.
+     * @param world the world.
+     */
+    private static void loadChunkAndNotify(final World world)
+    {
+        if(!ownedChunks.isEmpty())
+        {
+            final ChunkPos storage = ownedChunks.keySet().iterator().next();
+            loadChunk(world, storage.getX(), storage.getZ(), storage.getDim());
+        }
+        else if(!closeChunks.isEmpty())
+        {
+            final ChunkLoadStorage storage = closeChunks.get(0);
+            if(world.provider.getDimension() != storage.getDimension())
+            {
+                return;
+            }
+            final BlockPos idNow = storage.getPos();
+            if (removedElements % 100 == 0)
+            {
+                Log.getLogger().info("Loaded 100 chunks with close colony, now loading: " + idNow.getX() + ":" + idNow.getZ());
+            }
+            final int id = storage.getColonyId();
+            closeChunks.remove(0);
+            final Chunk chunk = world.getChunkFromChunkCoords(idNow.getX(), idNow.getZ());
+            final IColonyTagCapability cap = chunk.getCapability(CLOSE_COLONY_CAP, null);
+
+            if(storage.isAdd())
+            {
+                cap.addColony(id);
+            }
+            else
+            {
+                cap.removeColony(id);
+            }
+            MineColonies.getNetwork().sendToAll(new UpdateChunkCapabilityMessage(cap, chunk.x, chunk.z));
+            removedElements++;
+            chunk.markDirty();
+        }
+    }
+
+    /**
+     * Load the colony info for a certain chunk.
+     * @param world the world.
+     * @param x the x pos.
+     * @param z the z pos.
+     * @param dim the dimension.
+     */
+    public static void loadChunk(final World world, final int x, final int z, final int dim)
+    {
+        final ChunkPos pos = new ChunkPos(x, z, dim);
+        if(!ownedChunks.isEmpty() && ownedChunks.containsKey(pos))
+        {
+            final ChunkLoadStorage storage = ownedChunks.get(pos);
+            if(world.provider.getDimension() != storage.getDimension())
+            {
+                return;
+            }
+            if (removedElements % 100 == 0)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 100 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)